extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksContainerFactory.java

```diff
@@ -85,4 +135,38 @@ private String getAnnotationValue(AnnotationInstance annotationInstance, String
         }
         return value.asString();
     }
+
+    private boolean isCollection(Type type, IndexView index) {
+        if (type.kind() == Type.Kind.PRIMITIVE) {
+            return false;
+        }
+        ClassInfo classInfo = index.getClassByName(type.name());
+        if (classInfo == null) {
+            return false;
+        }
+        return classInfo.interfaceNames().stream().anyMatch(DotName.createSimple(Collection.class.getName())::equals);
+    }
+
+    private Type getNonAsyncReturnType(Type returnType) {
+        switch (returnType.kind()) {
+            case ARRAY:
+            case CLASS:
+            case PRIMITIVE:
+            case VOID:
+                return returnType;
+            case PARAMETERIZED_TYPE:
+                // NOTE: same code in RuntimeResourceDeployment.getNonAsyncReturnType
```

Comment:
Do you mean `getEffectiveReturnType`? If so, maybe we should reuse that by making it `static`?

Reply:
I copied this method from resteasy reactive, so yes, we should make it static and reuse it from both places.

Reply:
Can you please make this move in a different commit? Because if we have to revert it something, we don't want to revert the whole thing

Reply:
Yes, +100

Reply:
PR updated